static/js/tools/stat_var/page.tsx

```diff
@@ -167,12 +166,130 @@ class Page extends Component<unknown, PageStateType> {
     );
   }
 
-  private updateHash(sv: string): void {
-    window.location.hash = `#${sv}`;
+  /**
+   * Fetches sources to display in dropdown menu.
+   */
+  private async fetchSources(): Promise<void> {
+    axios
+      .get("/api/browser/propvals/typeOf/Source")
+      .then((resp) => {
+        const sourcePromises = [];
+        for (const source of resp.data?.values?.in) {
```

Comment:
I'm curious, when using "?", what happens if it's false? e.g., what will happen if resp.data? is false?

Reply:
oh hmm I tested and I think I probably shouldn't do this as it will not have a length to iterate over if missing, which will cause an error. I added an existence check instead